### ISSUE TITLE
Enhancement: Default allowAutomaticSingleRunInference to true

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#8121]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

Reposting #6176: we've heard from users that the `allowAutomaticSingleRunInference` option can significantly improve linting performance. And other than ~`--fix` mode~ project references (#3851) we haven't heard any new reports about it causing issues.

Let's turn the inference on by default!

### Additional Info

Marking as blocked on #3851 to start. Adding to the ~v7~ v8 milestone as well.

**Comment by liuxingbaoyu** [comment]
FYI: This has obvious benefits in Babel.
https://github.com/babel/babel/pull/16029

**Comment by JoshuaKGoldberg** [comment]
#8922 was merged into `v8`. ✅
